xrpl/models/transactions/credential_create.py

```diff
@@ -0,0 +1,76 @@
+"""Model for CredentialCreate transaction type."""
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+from typing_extensions import Self
+
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import (
+    HEX_REGEX,
+    KW_ONLY_DATACLASS,
+    _get_credential_type_error,
+    require_kwargs_on_init,
+)
+
+_MAX_URI_LENGTH = 256
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class CredentialCreate(Transaction):
+    """This transaction creates a Credential object. It must be sent by the issuer."""
+
+    transaction_type: TransactionType = field(
+        default=TransactionType.CREDENTIAL_CREATE,
+        init=False,
+    )
+
+    account: str = REQUIRED  # type: ignore
+    """
+    The issuer of the credential.
+    """
+    subject: str = REQUIRED  # type: ignore
+    """
+    The subject of the credential.
+    """
+
+    credential_type: str = REQUIRED  # type: ignore
+    """
+    A (hex-encoded) value to identify the type of credential from the issuer.
+    """
+    expiration: Optional[int] = None
+    """
+    Optional credential expiration.
+    """
+    uri: Optional[str] = None
+    """
+    Optional additional data about the credential (such as a link to the Verifiable
+    Credential document).
+    """
+
+    def _get_errors(self: Self) -> Dict[str, str]:
+        return {
```

Comment:
I still think this structure is less readable - I prefer the `if`s over the complexity of a lambda

Reply:
I like the structure of the lambda, it makes all the different validations clear. The cascading if-else statements are slightly less convinient.

Reply:
Yeah I agree with @mvadari it looks confusing to read.

Reply:
We have had several deliberations on the "readability" of a piece of a code. This is influenced by much of a developer's personal preferences.

In future, I hope we can allow for certain flexibility in the PR author's choice of code structure.

Reply:
An open-source library isn't a normal repo, it serves as documentation as well (for how to interact with these features). So readability trumps developer preference in this repo.